assembly/src/ast/mod.rs

```diff
@@ -1,107 +1,58 @@
 //! Abstract syntax tree (AST) components of Miden programs, modules, and procedures.
-//!
-//! Structs in this module (specifically [ProgramAst] and [ModuleAst]) can be used to parse source
-//! code into relevant ASTs. This can be done via their `parse()` methods.
-use super::{
-    crypto::hash::RpoDigest, ByteReader, ByteWriter, Deserializable, DeserializationError, Felt,
-    LabelError, LibraryPath, ParsingError, ProcedureId, ProcedureName, Serializable, SliceReader,
-    StarkField, Token, TokenStream, MAX_LABEL_LEN,
-};
-use crate::utils::{collections::*, string::*};
-use vm_core::utils::bound_into_included_u64;
 
 pub use tracing::{event, info_span, instrument, Level};
```

Comment:
I don't believe so, no. In fact, I would be inclined to import these under the `diagnostics` module instead, but since this pre-existed my changes, and I didn't have any compelling reason to touch it, I left it as-is. AFAIK, despite being `pub`, these are not imported from here downstream anywhere.

Reply:
The `tracing` APIs are now re-exported from the `diagnostics` module. Only `instrument` was actually being used, and entirely within the `assembly` crate, so definitely not a big deal moving it around.